server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -151,6 +154,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("qtp-query-" + queryId);
```

Comment:
a more elegant approach might be to use log4j's diagnostic contexts http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/NDC.html and http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/MDC.html